libs/features/mas/web-components/src/merch-datasource.js

```diff
@@ -148,60 +200,93 @@ const cache = new FragmentCache();
  */
 export class MerchDataSource extends HTMLElement {
     /**
-     * @type {import('./aem.js').AEM}
+     * @type {import('@adobe/mas-web-components').AEM}
      */
     #aem;
     cache = cache;
 
+    /**
+     * @type {HtmlElement[]}
+     */
+    refs = [];
+
     /**
      * @type {string} fragment path
      */
     path;
 
+    /**
+     * Consonant styling for CTAs.
+     */
+    consonant = false;
+
+    /**
+     * Internal promise to track the readiness of the web-component to render.
+     */
+    #readyPromise;
+
     static get observedAttributes() {
-        return ['source', 'path'];
+        return ['source', 'path', 'consonant'];
     }
 
     attributeChangedCallback(name, oldValue, newValue) {
         this[name] = newValue;
     }
 
     connectedCallback() {
+        this.consonant = this.hasAttribute('consonant');
+        this.clearRefs();
         const bucket =
             this.getAttribute(ATTR_AEM_BUCKET) ??
-            document.querySelector('mas-studio')?.getAttribute(ATTR_AEM_BUCKET);
+            'publish-p22655-e59341';
         this.#aem = new AEM(bucket);
-        this.fetchData();
+        this.refresh(true);
+    }
+
+    clearRefs() {
+        this.refs.forEach((ref) => {
+            ref.remove();
```

Comment:
what is the reasoning behind those refs + removal? to remove readd everything but datasource child?

Reply:
@npeltier MAS studio refreshes datasource un updates.  merch-card can have elements in the default slot that are not created by the datasource but by the consumer. e.g: mas studio adds a `sp-status-light` to display the publish status. Basically, when we refresh a datasource, we only clear and re-create the elements that are tracked.